### DISCUSSION TITLE
Error when compile TVision in Cygwin

### DISCUSSION THREAD

**Discussion by PauloES** [magiblot/tvision#69] [General]
Hi,

When a try to compile running the command below in Cygwin ocurrs the erro below.  Does anybody could help-me, please?

**usuario@DESKTOP-A8VT3IL ~/tvision-master/build
$ make**

Consolidate compiler generated dependencies of target tvision
[  0%] Building CXX object source/CMakeFiles/tvision.dir/cmake_pch.hxx.gch
In file included from /home/usuario/tvision-master/include/tvision/tstrview.h:12,
                 from /home/usuario/tvision-master/include/tvision/tv.h:650,
                 from /home/usuario/tvision-master/include/pch.h:3,
                 from /home/usuario/tvision-master/build/source/CMakeFiles/tvision.dir/cmake_pch.hxx:5,
                 from <command-line>:
/usr/include/string.h:130:10: error: conflicting declaration of ‘char* strupr(char*)’ with ‘C’ linkage
  130 | char    *strupr (char *);
      |          ^~~~~~
In file included from /home/usuario/tvision-master/include/tvision/tv.h:647,
                 from /home/usuario/tvision-master/include/pch.h:3,
                 from /home/usuario/tvision-master/build/source/CMakeFiles/tvision.dir/cmake_pch.hxx:5,
                 from <command-line>:
/home/usuario/tvision-master/include/tvision/util.h:91:7: note: previous declaration with ‘C++’ linkage
   91 | char *strupr(char *s) noexcept;
      |       ^~~~~~
In file included from /usr/lib/gcc/x86_64-pc-cygwin/11/include/c++/cstdlib:75,
                 from /usr/lib/gcc/x86_64-pc-cygwin/11/include/c++/ext/string_conversions.h:41,
                 from /usr/lib/gcc/x86_64-pc-cygwin/11/include/c++/bits/basic_string.h:6607,
                 from /usr/lib/gcc/x86_64-pc-cygwin/11/include/c++/string:55,
                 from /home/usuario/tvision-master/include/pch.h:6,
                 from /home/usuario/tvision-master/build/source/CMakeFiles/tvision.dir/cmake_pch.hxx:5,
                 from <command-line>:
/usr/include/stdlib.h:227:9: error: conflicting declaration of ‘char* itoa(int, char*, int)’ with ‘C’ linkage
  227 | char *  itoa (int, char *, int);
      |         ^~~~
In file included from /home/usuario/tvision-master/include/tvision/tv.h:647,
                 from /home/usuario/tvision-master/include/pch.h:3,
                 from /home/usuario/tvision-master/build/source/CMakeFiles/tvision.dir/cmake_pch.hxx:5,
                 from <command-line>:
/home/usuario/tvision-master/include/tvision/util.h:92:7: note: previous declaration with ‘C++’ linkage
   92 | char *itoa( int value, char *buffer, int radix ) noexcept;
      |       ^~~~
make[2]: *** [source/CMakeFiles/tvision.dir/build.make:77: source/CMakeFiles/tvision.dir/cmake_pch.hxx.gch] Error 1
make[1]: *** [CMakeFiles/Makefile2:263: source/CMakeFiles/tvision.dir/all] Error 2
make: *** [Makefile:136: all] Error 2

**Comment by magiblot** [comment]
Hello,

This is not your fault, I had never attempted to build Turbo Vision in Cygwin. I'll see what I can do.

Cheers.

**Reply by magiblot** [reply]
Cygwin will not be supported for the time being. If you are on Windows, note that you can also build Turbo Vision with Visual Studio or in WSL.

**Reply by PauloES** [reply]
Thanks for your quickly response! I will try Visual Studio. Thanks a lot and congratulation for your initiative.

Best regards

**Reply by magiblot** [reply]
To build with Visual Studio, just install Visual Studio Community and CMake for Windows, then run CMake from the command line. This is probably simpler than building through the Visual Studio UI.
